exporters/otlp/profiles/src/main/java/io/opentelemetry/exporter/otlp/profiles/DictionaryTable.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.profiles;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This data structure is effectively an indexed Set.
+ *
+ * <p>It stores each offered element without duplication (like a Set, not a List), but supports
+ * reference to elements by their position (like a List index).
+ *
+ * <p>This class is not threadsafe and must be externally synchronized.
+ *
+ * <p>For a given Object o, after i = putIfAbsent(o), then getTable().get(i).equals(o);
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
```

Comment:
this class isn't in an internal package, and hence shouldn't have this notation.